RecFCCeeCalorimeter/src/components/CaloTowerToolFCCee.h

```diff
@@ -0,0 +1,219 @@
+#ifndef RECFCCEECALORIMETER_CALOTOWERTOOLFCCEE_H
+#define RECFCCEECALORIMETER_CALOTOWERTOOLFCCEE_H
+
+// from Gaudi
+#include "GaudiAlg/GaudiTool.h"
+
+// k4geo
+#include "detectorSegmentations/FCCSWGridModuleThetaMerged_k4geo.h"
+
+// k4FWCore
+#include "k4FWCore/DataHandle.h"
+#include "k4Interface/ITowerToolThetaModule.h"
+class IGeoSvc;
+
+// dd4hep
+#include "DDSegmentation/MultiSegmentation.h"
+
+namespace dd4hep {
+namespace DDSegmentation {
+class Segmentation;
+class BitFieldCoder;
+}
+}
+
+// edm4hep
+namespace edm4hep {
+class CalorimeterHitCollection;
+class CalorimeterHit;
+class Cluster;
+}
+
+
+/** @class CaloTowerToolFCCee Reconstruction/RecFCCeeCalorimeter/src/components/CaloTowerToolFCCee.h
+ *
+ *  Tool building the calorimeter towers for the sliding window algorithm.
+ *  This tool runs over all calorimeter systems (ECAL barrel, HCAL barrel + extended barrel, calorimeter endcaps,
+ * forward calorimeters). If not all systems are available or not wanted to be used, create an empty collection using
+ * CreateDummyCellsCollection algorithm.
+ *  Towers are built of cells in theta-phi, summed over all radial layers.
+ *  A tower contains all cells within certain theta and phi (tower size: '\b deltaThetaTower', '\b deltaPhiTower').
+ *
+ *  @author Anna Zaborowska
+ *  @author Jana Faltova
+ *  @Modified by Tong Li, for Theta-Module Merged readouts in FCCee
+ */
+
+class CaloTowerToolFCCee : public GaudiTool, virtual public ITowerToolThetaModule {
+public:
+  CaloTowerToolFCCee(const std::string& type, const std::string& name, const IInterface* parent);
+  virtual ~CaloTowerToolFCCee() = default;
+  /**  Initialize.
+   *   @return status code
+   */
+  virtual StatusCode initialize() final;
+  /**  Finalize.
+   *   @return status code
+   */
+  virtual StatusCode finalize() final;
+  /**  Find number of calorimeter towers.
+   *   Number of towers in phi is calculated from full azimuthal angle (2 pi) and the size of tower in phi ('\b
+   * deltaPhiTower').
+   *   Number of towers in theta is calculated from maximum detector theta ('\b thetaMax`) and the size of tower in theta ('\b
+   * deltaThetaTower').
+   *   @return Array containing number of towers in theta and phi.
+   */
+  virtual tower towersNumber() final;
```

Comment:
this tower struct needs to go :)
See https://github.com/key4hep/k4FWCore/pull/175

Reply:
Done